sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1305,7 +1305,13 @@ def _apply_migrations(
         versions = self.get_versions(validate=False)
         migrations = MIGRATIONS[versions.schema_version :]
 
-        migrate_rows = migrations or major_minor(SQLGLOT_VERSION) != versions.minor_sqlglot_version
+        migrate_rows = any(
+            [
+                migrations,
+                major_minor(SQLGLOT_VERSION) != versions.minor_sqlglot_version,
+                major_minor(SQLMESH_VERSION) != versions.minor_sqlmesh_version,
```

Comment:
What is the motivation for this change? The sqlmesh minor version was omitted intentionally to avoid migration when it's not needed even if the minor version was bumped. Generally there are only 2 reasons to migrate: if there's a new migration script (the `migrations` collection) or if the sqlglot version changed.

Reply:
I read the issue carefully and now I understand the motivation better. However, `migrate_rows` not only controls whether the backups are made but also whether the migration should be performed. I suggested decoupling the 2 and only do backups on sqlmesh minor versions changes without affecting the `migrate_rows` flag.

Reply:
After reading your comments and digging in to the code a bit more I believe I can see what you're talking about.

Hopefully I've addressed your concerns by only performing a table copy when it's a sqlmesh minor change without impacting the flag that drives the complex snapshot and environment migration process.

Reply:
Yes, this looks great, thank you! However, I'm not sure why the refactor was performed in the other method (`def migrate`)? Is that change necessary? If not, can we please revert it.